interface/main/holidays/Holidays_Controller.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * interface/main/holidays/Holidays_Controller.php implementation of holidays logic.
+ *
+ * This class contains the implementation of all the logic
+ * included in the holidays calendar story.
+ *
+ * Copyright (C) 2016 Sharon Cohen <sharonco@matrix.co.il>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  sharonco <sharonco@matrix.co.il>
+ * @link    http://www.open-emr.org
+ */
+
+require_once("Holidays_Storage.php");
+class Holidays_Controller{
+
+    const UPLOAD_DIR = "documents/holidays_storage";
+    const FILE_NAME = "holidays_to_import.csv";
+
+    
+    public $storage;
+    public $target_file;
+
+    function __construct(){
+        $this->set_target_file();
+        $this->storage = new Holidays_Storage();
+    }
+
+    public function set_target_file(){
+            $this->target_file = $GLOBALS['OE_SITE_DIR']."/". self::UPLOAD_DIR."/".self::FILE_NAME;
+    }
+    public function get_target_file(){
+        return $this->target_file;
+    }
+
+    /**
+     * This function uploads the csv file
+     * @param $files
+     * @return bool
+     */
+    public function upload_csv($files){
+        $file_type = pathinfo($this->target_file,PATHINFO_EXTENSION);
+        if($file_type != "csv"){
+            return false;
+        }
```

Comment:
If the holidays_storage directory does not yet exist, then I think you will need to create it here.